src/cmd/new.rs

```diff
@@ -27,8 +27,8 @@ pub(crate) fn execute<P>(name: &str, dir: Option<P>) -> Result<Option<String>>
 where
     P: AsRef<Path>,
 {
-    if name.contains('-') {
-        anyhow::bail!("Contract names cannot contain hyphens");
+    if !name.chars().all(|c| c.is_alphanumeric() || c == '_') {
```

Comment:
this allows `_` as identifier which is invalid. however, I think this isn't too bad?

Reply:
Well spotted. Is that even a valid crate name?

Reply:
I just checked and `cargo` does not immediately put a stop to having `_` as crate name but my filesystem definitely does. xD

Reply:
Note sure we really need to guard against this. What is your opinion?

Reply:
Seems unnecessary